src/ln/msgs.rs

```diff
@@ -1200,112 +1265,76 @@ impl<R: Read> Readable<R> for UnsignedNodeAnnouncement {
 		r.read_exact(&mut rgb)?;
 		let alias: [u8; 32] = Readable::read(r)?;
 
-		let addrlen: u16 = Readable::read(r)?;
-		let mut addr_readpos = 0;
-		let mut addresses = Vec::with_capacity(4);
-		let mut f: u8 = 0;
-		let mut excess = 0;
-		loop {
-			if addrlen <= addr_readpos { break; }
-			f = Readable::read(r)?;
-			match f {
-				1 => {
-					if addresses.len() > 0 {
-						return Err(DecodeError::ExtraAddressesPerType);
-					}
-					if addrlen < addr_readpos + 1 + 6 {
-						return Err(DecodeError::BadLengthDescriptor);
-					}
-					addresses.push(NetAddress::IPv4 {
-						addr: {
-							let mut addr = [0; 4];
-							r.read_exact(&mut addr)?;
-							addr
+		let addr_len: u16 = Readable::read(r)?;
+		let mut addresses: Vec<NetAddress> = Vec::with_capacity(addr_len as usize);
```

Comment:
This with_capacity is gonna let an attacker use way more memory than they should be able to, though 64k isnt so bad.

Reply:
From where  do you get 64k ? Rust's GlobalAlloc use libc and this one doesn't enforce a hard limit beyond what mmap can get you ?
Should we hardcode a reasonable limit to avoid system resource abuse ?

Reply:
Oh, right, its 64K entries (u16) but the Vec entries are much higher. Anyway, given this Vec can never be longer than 4 entries or so, just with_capacity(4) or new() are sufficient.